crawl-ref/source/spl-summoning.cc

```diff
@@ -948,10 +949,14 @@ spret_type cast_summon_lightning_spire(int pow, const coord_def& where, god_type
         fail_check();
 
         // invisible monster
-        mpr("Something you can't see is blocking your construction!");
+        canned_msg(MSG_GHOSTLY_OUTLINE);
         return SPRET_SUCCESS;
     }
 
+    spret_type handle_trap = handle_trap_at_target_location(where, fail);
+    if (handle_trap != SPRET_NONE)
+        return handle_trap;
+
```

Comment:
is this the appropriate way to do this? It feels a bit wonky, but my intent was:

1. see if we need to handle traps at the target location.
2. if so, it will return something other than SPRET_NONE, so anything else we get we gotta return to the function above this.

I don't know if there's a better C++ pattern for something like this.